src/index.ts

```diff
@@ -1,70 +1,84 @@
 import debounce from 'lodash/debounce';
 import throttle from 'lodash/throttle';
 
+type EventType = keyof GlobalEventHandlersEventMap;
+
+type EventData = {
+  listeners: Record<string, HandlerFunction>;
+  lastIndex: number;
+  listenersLength: number;
+};
+
+type EventSettings = {
+  emitter?: Document | typeof globalThis;
+  wrapper?: (callback: Function) => Function;
```

Comment:
We should avoid anything similar to `any`. `Function` is basically `any` function.

It's practically to tell the structure of `Function` as it's generic, so type-safety is incomplete.

A fix would be to define types to explicitly describe your function.